shell/platform/darwin/ios/BUILD.gn

```diff
@@ -227,6 +227,7 @@ source_set("ios_test_flutter_mrc") {
     "-F$platform_frameworks_path",
     "-mios-simulator-version-min=$ios_testing_deployment_target",
   ]
+  cflags_objcc = [ "-Wno-unguarded-availability-new" ]
```

Comment:
Please add a comment explaining why this is here, and what the conditions are for getting rid of it (i.e., the XCTest header bug being fixed).